core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -384,7 +384,7 @@ package object config {
         "get the replication level of the block to the initial number")
       .version("2.2.0")
       .booleanConf
-      .createWithDefault(false)
+      .createWithDefault(true)
```

Comment:
Should we maybe just enable this by default when we're in Kubernates? I am okay with enabling it by default too if other people are fine. cc @tgravescs and @Ngone51 too FYI

Reply:
For the other resource managers, this will be helpful because this is a kind of self-healing code.

Reply:
I am not sure how widely this is used, particularly as it is not enabled by default.
Especially in context of dynamic resource allocation, it can become very chatty when executor's start getting dropped.

Given this, I am not very keen on enabling it atleast for yarn. Thoughts @tgravescs ?

Reply:
Hi, @mridulm . Actually, we are using it now and it's a good time to test it by default, isn't it?
> I am not sure how widely this is used, particularly as it is not enabled by default.

For the following, Apache Spark usually drop only empty executors. If you are saying a storage timeout configuration, I believe that what we need is to improve that configuration behavior after this enabling. I guess `storage timeout` should not cause any `chatty` situation.
> Especially in context of dynamic resource allocation, it can become very chatty

Reply:
Thanks for the ping. I think I'm OK with the change. And shall we document the behaviour change in `core-migration-guide.md`?

Reply:
Sure, I'll update the PR, @Ngone51 .

Reply:
In the past, I found this to be noisy for the cases where replication was enabled - but this was a while back, and I would like to understand better what the 'cost' of enabling this for nontrivial usecases is for master : disabled by default means only developers who specifically test for it pay the price; not everyone.
It is quite common for an application to have references to a persisted RDD even after its use - with the loss of the RDD blocks having little to no functional impact.
This is similar to loss of blocks for an unreplicated persisted RDD - we do not proactively recompute the lost blocks; but do so on demand.

If the idea is we enable this for master, and evaluate the impact over the next 6 months and revisit at the end, I am fine with that: but an evaluation would need to be done before this goes out - else anyone using replicated storage will also get hit with the impact of proactive replication as well, and will need to disable this for their applications.

Reply:
> but an evaluation would need to be done before this goes out 

or perhaps identify the subset of conditions where it makes sense to enable it by default.